src/tunnel/server.rs

```diff
@@ -94,7 +95,22 @@ async fn run_tunnel(
             static SERVERS: Lazy<Mutex<HashMap<(Host<String>, u16), mpsc::Receiver<TcpStream>>>> =
                 Lazy::new(|| Mutex::new(HashMap::with_capacity(0)));
 
-            let local_srv = (remote.host, remote.port);
+            // Determine if the requested port is to be mapped to a different port.
+            let remote_port = restriction.allow
+                .iter()
+                .find_map(|allow| {
+                    if let AllowConfig::ReverseTunnel(allow) = allow {
+                        if remote.protocol.is_reverse_tunnel() {
```

Comment:
No need of this check, as you already know at this point that a ReverseTcp has been requested.

Reply:
Done!